### PR TITLE
Add extra logs when sending a deposit and don't allow the user to exceed their balance

### PR DESCRIPTION
**Pull request by hozzjss**
Fixes:
- https://github.com/stacks-network/sbtc/issues/1137
- https://github.com/stacks-network/sbtc/issues/1141

### CODE REVIEW

File: src/comps/Deposit.tsx [hunk 1]
```diff
@@ -75,25 +78,38 @@ type DepositFlowStepProps = {
...
-      .number()
-      // dust amount is in sats
-      .min(
-        minDepositAmount,
-        `Minimum deposit amount is ${minDepositAmount} BTC`,
-      )
-      .max(maxDepositAmount, `Current deposit cap is ${maxDepositAmount} BTC`)
-      .required(),
-  });
+  let minDepositAmount = perDepositMinimum / 1e8;
```
Comment (matteojug) [inline src/comps/Deposit.tsx:91]: nit: can be const

File: src/comps/Deposit.tsx [hunk 2]
```diff
@@ -511,11 +547,27 @@ const DepositFlow = () => {
...
+    },
+    enabled: !!btcAddress,
+  });
   const renderStep = () => {
     switch (step) {
       case DEPOSIT_STEP.AMOUNT:
         return (
-          <DepositFlowAmount setAmount={setAmount} setStep={handleUpdateStep} />
+          <DepositFlowAmount
+            btcBalance={btcBalance || 0}
```
Comment (matteojug) [inline src/comps/Deposit.tsx:567]: IIUC defaulting to `0` means that if we are having troubles reaching out to the mempool api, users will not be able to use the bridge. If so, should we use a `MAX_MONEY` default instead (or disable validation if we cannot get the balance)? Alternatively, we could change the check wrt the user balance to be just a fyi kind of thing, not blocking the flow -- to that if we/api mess up the balance computation, the user can still go ahead with the process.

File: src/hooks/use-mint-caps.ts [hunk 3]
```diff
@@ -38,9 +38,11 @@ export default function useMintCaps() {
       if (!currentEmilyLimitsData || !currentSbtcSupplyData) {
         return false;
       }
-      const withinMintCap =
-        currentEmilyLimitsData.pegCap >
+      const supplyPlusDeposit =
         Number(currentSbtcSupplyData.value.value) + amount;
+      const withinMintCap =
+        amount > currentEmilyLimitsData.perDepositMinimum &&
+        supplyPlusDeposit < currentEmilyLimitsData.pegCap;
```
Comment (matteojug) [inline src/hooks/use-mint-caps.ts:45]: Should we use `>=` and `<=` here?

File: src/comps/Deposit.tsx [hunk 4]
```diff
@@ -281,6 +297,15 @@ const DepositFlowConfirm = ({
...
           network: walletNetwork,
         };
+        console.log({
+          preSendParams: {
+            bitcoinTxid: txId,
+            bitcoinTxOutputIndex: 0,
+            reclaimScript: reclaimScriptHex,
+            depositScript: depositScriptHexPreHash,
+          },
+        });
```
Comment (aldur) [inline src/comps/Deposit.tsx:307]: Can you make this a JSON? Thanks!

**Review by matteojug — APPROVED** [review]

**Pull request merged**